botocore/paginate.py

```diff
@@ -324,8 +324,8 @@ def search(self, expression):
             if isinstance(results, list):
                 for element in results:
                     yield element
-            else:
-                # Yield result directly if it is not a list.
+            elif results:
+                # Yield result directly if it is not a list and there are results.
```

Comment:
What if the result evaluates to `False` but is not `None`?

```suggestion
            elif results is None:
                return
            else:
                # Yield result directly if it is not a list and there are results.
```